core/base/src/TApplication.cxx

```diff
@@ -628,21 +630,416 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+/// TFunction lookup.
+///
+/// \param[in] className the name of the class
+/// \param[in] func pointer to the method
+
+static TString FormatReturnTypeForDoxygen(const TString &className, TFunction *func)
 {
-   gInterpreter->ProcessLine(line);
+   // We put the return type of "func" in a TString "returnType".
+   TString returnType = func->GetReturnTypeName();
+   // If the return type consist of enumeration, it will appear with its class (Class::Enumeration).
+   // Below we make sure to remove the current class, because the syntax of Doxygen requires it.
+   TString classNameRE ("\\b");
+   classNameRE.Append(className);
+   classNameRE.Append("::\\b");
+   TPRegexp returnFix(classNameRE);
+   returnFix.Substitute(returnType, "");
+   // We check is if the method is deffined as inline (inliceCheck).
+   Int_t inlineCheck = (func->ExtraProperty() & kIsInlined);
+   // inlineCheck is not zero only if the function is defined as inline.
+   if (inlineCheck == 0) {
+      // If the function is not inline we only change the spacing in "returnType".
+      returnType.ReplaceAll("*", " *");
+   } else {
+      // We check if the function is defined as virtual.
+      Int_t virtualCheck = (func->Property() & kIsVirtual);
+      if (virtualCheck != 0) {
+         // If the function is virtual, we append "virtual" before the return type.
+         TString buffer("virtual ");
+         buffer.Append(returnType);
+         returnType.Clear();
+         returnType.Append(buffer);
+      }
+      returnType.ReplaceAll(" *", "*");
+   }
+   // In any case (with no respect to virtual/inline check) we need to change
+   // the return type as following.
+   returnType.ReplaceAll("istream", "std::istream");
+   returnType.ReplaceAll("ostream", "std::ostream");
+   returnType.ReplaceAll("&", " &");
+   return returnType;
+}
+}
+
+namespace {
+////////////////////////////////////////////////////////////////////////////////
+/// The function generates in a URL for "enumeratorName" defined in "className".
+/// It returns a TString with the URL used in the online reference guide,
+/// generated with Doxygen. The URL generation is specific. It could be
+/// separated into 3 parts - URL for "className", part for the enumeration
+/// and a part for the enumerator.
+///
+/// \param[in] className the name of the class
+/// \param[in] enumeratorName the name of the enumerator from the class
+/// \param[in] dataMember pointer to the enumerator
+/// \param[in] scopeType enumerator to the scope type
 
-   Printf("\nROOT special commands.");
-   Printf("===========================================================================");
-   Printf("             pwd          : show current directory, pad and style");
-   Printf("             ls           : list contents of current directory");
-   Printf("             which [file] : shows path of macro file");
+static TString URLforEnumerator(TString &className, const TString &enumeratorName, TDataMember *dataMember, EUrl scopeType)
```

Comment:
Does the caller need to get the updated value of `className`? If not, use `TString className` - i.e. we want to copy the value into the function.